src/typing/typeloadFields.ml

```diff
@@ -139,7 +139,9 @@ let get_struct_init_super_info ctx c p =
 			let args = (try get_method_args ctor with Not_found -> []) in
 			let tl,el =
 				List.fold_left (fun (args,exprs) (v,value) ->
-					let opt = match value with Some _ -> true | None -> false in
+					let opt = match value with
+						| Some _ -> true
+						| None -> Meta.has Meta.Optional v.v_meta in
```

Comment:
Our coding style would put the `in` on a new line by itself. I realize that this makes real functional programmers cringe, but that's a sacrifice I'm willing to make.